src/components/lib/protdom/struct_defs_arch/default/protdom_stuct_def.h

```diff
@@ -0,0 +1,14 @@
+
+#include <cos_types.h>
+#include <protdom.h>
+#include <cos_kernel_api.h>
+
+/* we need to expose these definitions for the static slab allocators in the booter */
```

Comment:
Great comment.

Reply:
I wonder if you've tried to run the code without the x86-64 stuff enabled. Would be great if it works given you added the code for the default!